test/e2e/load.go

```diff
@@ -0,0 +1,143 @@
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2e
+
+import (
+	"fmt"
+	"math/rand"
+	"sync"
+	"time"
+
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/client"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/fields"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/labels"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/util"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+const (
+	image          = "gcr.io/google_containers/serve_hostname:1.1"
+	simulationTime = 5 * time.Minute
+	smallRCSize    = 5
+	mediumRCSize   = 30
+	bigRCSize      = 250
+)
+
+// This test suite can take a long time to run, so by default it is added to
+// the ginkgo.skip list (see driver.go).
+// To run this suite you must explicitly ask for it by setting the
+// -t/--test flag or ginkgo.focus flag.
+var _ = Describe("Load", func() {
+	var c *client.Client
+	var minionCount int
+	var ns string
+
+	BeforeEach(func() {
+		var err error
+		c, err = loadClient()
+		expectNoError(err)
+		minions, err := c.Nodes().List(labels.Everything(), fields.Everything())
+		expectNoError(err)
+		minionCount = len(minions.Items)
+		Expect(minionCount).NotTo(BeZero())
+		nsForTesting, err := createTestingNS("load", c)
+		ns = nsForTesting.Name
+		expectNoError(err)
+	})
+
+	AfterEach(func() {
+		By(fmt.Sprintf("Destroying namespace for this suite %v", ns))
+		if err := c.Namespaces().Delete(ns); err != nil {
+			Failf("Couldn't delete ns %s", err)
+		}
+	})
+
+	type Load struct {
+		skip          bool
+		podsPerMinion int
+	}
+
+	loadTests := []Load{
+		{podsPerMinion: 30, skip: true},
+	}
+
+	for _, testArg := range loadTests {
+		name := fmt.Sprintf("should be able to handle %v pods per minion", testArg.podsPerMinion)
+		if testArg.skip {
+			name = "[Skipped] " + name
+		}
+
+		It(name, func() {
+			totalPods := testArg.podsPerMinion * minionCount
+			smallRCCount, mediumRCCount, bigRCCount := computeRCCounts(totalPods)
+			threads := smallRCCount + mediumRCCount + bigRCCount
+
+			var wg sync.WaitGroup
+			wg.Add(threads)
+
+			// Run RC load for all kinds of RC.
+			runRCLoad(c, &wg, ns, smallRCSize, smallRCCount)
+			runRCLoad(c, &wg, ns, mediumRCSize, mediumRCCount)
+			runRCLoad(c, &wg, ns, bigRCSize, bigRCCount)
+
+			// Wait for all the pods from all the RC's to return.
+			wg.Wait()
+		})
+	}
+})
+
+func computeRCCounts(total int) (int, int, int) {
+	// Small RCs owns ~0.5 of total number of pods, medium and big RCs ~0.25 each.
+	bigRCCount := total / 4 / bigRCSize
```

Comment:
I agree that current implementation is a little bit confusing, but adding share factor will require converting the numbers between float and int representation.

Also the current implementation guarantees that the total number of pods will be (almost) equal to the requested "total".